### PR TITLE
Wired up events search

### PR DESCRIPTION
**Pull request by mxsnyder**
Per usual, ignore the "dist" folder.

### CODE REVIEW

File: js/apps/events-page/controllers/eventsPageCtrl.js [hunk 1]
```diff
@@ -213,13 +213,18 @@
 		angular.element(document).on('hide.bs.collapse', '.expando-wrapper .collapse', toggleIcon);
 		angular.element(document).on('show.bs.collapse', '.expando-wrapper .collapse', toggleIcon);
 
-		eventsService
-			.get(requestModel)
-			.then(processEvents)
-			.catch(handleFailedEventsGetRequest);
+		if ($location.search().term && $location.search().term.length) {
```
Comment (martypowell) [inline js/apps/events-page/controllers/eventsPageCtrl.js]: would this throw an error if term doesn't exist?

File: js/site-search.js [hunk 2]
```diff
@@ -62,7 +77,7 @@ bcpl.siteSearch = (($, window, constants) => {
 	$(document).on('click', siteSearchSearchIconSelector, onSearchIconClick);
 	$(document).on('click', searchButtonCatalogSelector, onSearchCatalogClick);
 	$(document).on('keyup', siteSearchInputSelector, onSiteSearchKeyup);
-	// $(document).on('click', searchButtonEventsSelector, onSearchEventsClick);
+	$(document).on('click', searchButtonEventsSelector, onSearchEventsClick);
 	// $(document).on('click', searchButtonWebsiteSelector, onSearchWebsiteClick);
```
Comment (martypowell) [inline js/site-search.js]: remove?

File: js/site-search.js [hunk 3]
```diff
@@ -44,8 +48,19 @@ bcpl.siteSearch = (($, window, constants) => {
...
+		}
+	};
+
+	const searchEvents = (activeWindow) => {
+		const searchTerms = getSearchTerms();
+
+		if (searchTerms.length) {
+			const baseWebsiteUrl = constants.baseWebsiteUrl;
+			const searchUrl = constants.search.urls.events;
+			activeWindow.location.href = `${baseWebsiteUrl}${searchUrl}${searchTerms}`; // eslint-disable-line
```
Comment (martypowell) [inline js/site-search.js]: in this case, search terms contains the ? for query params?

File: js/site-search.js [hunk 4]
```diff
@@ -44,8 +48,19 @@ bcpl.siteSearch = (($, window, constants) => {
 		const searchTerms = getSearchTerms();
 
 		if (searchTerms.length) {
```
Comment (martypowell) [inline js/site-search.js]: think you should either modfiy the getSearchTerms function to return an empty array or string if there are no search terms, or check the exists of searchTerms in the conditional

File: js/apps/events-page/controllers/eventsPageCtrl.js (hunk 1 above)
Comment (mxsnyder) [reply js/apps/events-page/controllers/eventsPageCtrl.js]: No - if `term` doesn't exist, it'll just be undefined and fail the `if`.

File: js/apps/events-page/controllers/eventsPageCtrl.js [hunk 5]
```diff
@@ -97,7 +97,7 @@
 
 		const handleFailedEventsGetRequest = (error) => {
 			self.isLoading = false;
-			self.requestErrorMessage = "There was a problem retrieving events. Please try again later.";
+			self.requestErrorMessage = 'There was a problem retrieving events. Please try again later.';
```
[the remark this answers is not in the document]
Comment (mxsnyder) [reply js/apps/events-page/controllers/eventsPageCtrl.js]: Noted.

File: js/site-search.js (hunk 2 above)
Comment (mxsnyder) [reply js/site-search.js]: I'm going to leave it since it's getting used in another branch that will succeed this one.

Comment (martypowell) [reply js/site-search.js]: can you just thrown in a comment saying that?

File: js/site-search.js (hunk 3 above)
Comment (mxsnyder) [reply js/site-search.js]: Correct.

File: js/site-search.js (hunk 4 above)
Comment (mxsnyder) [reply js/site-search.js]: Done.

**Comment by mxsnyder** [comment]
Ready for another look.

**Pull request merged**